lib/jekyll/page.rb

```diff
@@ -39,8 +39,10 @@ def initialize(site, base, dir, name)
       @base = base
       @dir  = dir
       @name = name
-      @path = if site.in_theme_dir(base) == base # we're in a theme
-                site.in_theme_dir(base, dir, name)
+
+      theme = site.theme_with_root(base)
+      @path = if theme # we're in a theme
```

Comment:
Unnecessary local variable initilization. Inlining is fine:
```suggestion
      @path = if site.theme_with_root(base) # we're in a theme
```